### DISCUSSION TITLE
(Tutorial) LocalDoH For ESNI / ECH support (for windows *maybe)

### DISCUSSION THREAD

**Discussion by NoUsernameinhere** [AdguardTeam/AdGuardHome#5154] [General]
idk this already known or not so here this..

first make your own cert , cause this for own `localhost` so you can read this guide [creating-your-own-certification](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Local-DoH#creating-your-own-certification)
after make your own cert to you adguardhome and move to Encryption settings like this i use default settings
![image](https://user-images.githubusercontent.com/117507125/202216526-6752f327-2abc-4c57-ae00-11d3c466f312.png)
![image](https://user-images.githubusercontent.com/117507125/202216506-16dedca6-17d0-41cf-84db-a4f7f6c3aeed.png)
locate and put cert and key file path

mine server like this `https://127.0.0.1/dns-query` put this on browser like brave , edge or you like
and don't forget turn on in flag setting  for Chromium based _execpt_ ms edge Chromium there separated setting for it
`#dns-https-svcb`
`#encrypted-client-hello`
restart browser

on firefox can read this [how-to-enable-esni-in-firefox](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Local-DoH#how-to-enable-esni-in-firefox)

After that you can check to this site [defo](https://defo.ie/ech-check.php) and [cloudflare](https://www.cloudflare.com/ssl/encrypted-sni/#results)

**Comment by xenstar** [comment]
I tried this method it didn't work for me.. because Adguard Home natively don't support it yet..
